tracer/test/Datadog.Trace.TestHelpers/EnvironmentHelper.cs

```diff
@@ -426,5 +452,12 @@ public string GetTargetFramework()
 
             return $"net{_major}{_minor}{_patch ?? string.Empty}";
         }
+
+        public MockTracerAgent GetMockAgent(bool useStatsD = false)
```

Comment:
Consider moving this into a static `MockTracerAgent.Create()` instead, or adding a ctor overload that doesn't take a port and just finds one automatically. This way `EnvironmentHelper` and `MockTracerAgent` are not tied in any way.

Reply:
The transport tests rely on the information from the environment helper and making the change in one method was far more palatable than making the changes in every single test to accept parameters from the environment helper. The idea was to have the environment helper determine what the agent is initialized with per the environment.
This is some sample code:
```
        public MockTracerAgent GetMockAgent()
        {
            MockTracerAgent agent;
            AgentPort = TcpPortProvider.GetOpenPort();
            _output.WriteLine($"Assigning port {AgentPort} for the agentPort.");
#if NETCOREAPP
            if (TransportType == TestTransports.Uds)
            {
                agent = new MockTracerAgent(traceUdsName: TraceUdsName, statsUdsName: StatsUdsName);
            }
            else
            {
                // Default
                agent = new MockTracerAgent(AgentPort);
            }
#else
            agent = new MockTracerAgent(AgentPort);
#endif

            return agent;
        }
```

If you have strong opinions about it living in the mock tracer agent, and the mock tracer agent accepting the environment helper in the constructor, I can do that. 
This seemed fitting to me, but the goal is to control the strategy based on the environment helper.

Reply:
I'm assuming that the intention is to only run a subset of tests on Uds right? In which case, it seems to make sense to just do that in the test directly? 🤔

```csharp
using var agent = new MockTracerAgent(traceUdsName: TraceUdsName, statsUdsName: StatsUdsName);
```

If we're planning on running _all_ the tests on Uds as well then these changes seem to make sense, but I assume that's not the plan?

Reply:
I do plan to run all tests on UDS (and all tests on windows named pipes).
Those paths need to be tested just as thoroughly as TCP in my opinion.
Even if they don't run for every test, the difference in enabling the transport would be something along these lines for a test:
```
EnvironmentHelper.SetTransport(TransportTypes.Uds);
```
Which could be put in attributes or something:
```
[Transports(TransportTypes.Tcp, TransportTypes.Uds, TransportTypes.WindowsNamedPipe)]
public void SomeTest()
```

Reply:
>  I do plan to run all tests on UDS (and all tests on windows named pipes).

😟 That could significantly increase our testing time without a huge amount of benefit IMO. 

Obviously agree it needs to be thoroughly tested, but we're testing communication between the tracer and the agent - I don't see that running (for example) all the Kafka, RabbitMQ, Sql etc tests using TCP send _and_ UDS actually gives us any more coverage than just doing UDS for one of those. I'd rather have dedicated tests that explicitly exercise multiple edge cases where possible (e.g. missing agent etc) and a smoke test over using Uds everywhere?

Reply:
I'm not saying that it should run for every CI run, but I'd like to run it on a schedule, and have the option to turn it on for all the tests without code changes.

Reply:
Thanks for clarifying, Colin. `EnvironmentHelper.GetMockAgent()` as a helper to create mock agents sounds good as long as any agent state remains in the agents themselves. I'd like it if we could still create `MockTraceAgent` instances directly without the need for `EnvironmentHelper` to keep those two things sepatate.

Reply:
My concern here is addressed, thanks. I won't resolve the thread since there is a parallel conversation about CI run times.